gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/swagger/converter/api/OAIToAPIConverter.java

```diff
@@ -292,65 +294,61 @@ protected SwaggerApiEntity fill(final SwaggerApiEntity apiEntity, OpenAPI oai) {
+
+                                                if (policy.isPresent()) {
+                                                    final Rule rule = new Rule();
+                                                    rule.setEnabled(true);
+                                                    rule.setDescription(
+                                                        operation.getSummary() == null
+                                                            ? (
+                                                                operation.getOperationId() == null
+                                                                    ? operation.getDescription()
+                                                                    : operation.getOperationId()
+                                                            )
+                                                            : operation.getSummary()
+                                                    );
+                                                    rule.setMethods(singleton(HttpMethod.valueOf(httpMethod.name())));
+
+                                                    io.gravitee.definition.model.Policy defPolicy = new io.gravitee.definition.model.Policy();
+                                                    defPolicy.setName(policy.get().getName());
+                                                    defPolicy.setConfiguration(clearNullValues(policy.get().getConfiguration()));
+                                                    rule.setPolicy(defPolicy);
+                                                    rules.add(rule);
+                                                }
                                             }
-                                        }
-                                    )
-                        );
-                        paths.put(path, rules);
+                                        )
+                            );
+                            paths.put(path, rules);
+                        }
                     }
                 );
 
-            apiEntity.setPaths(paths);
+            if (swaggerDescriptor.isWithPolicyPaths()) {
+                apiEntity.setPaths(paths);
+            }
         }
 
         // Path Mappings
-        if (apiEntity.getPaths() != null) {
-            apiEntity.setPathMappings(apiEntity.getPaths().keySet());
-        }
-
-        final String defaultDeclaredPath = "/";
-        Map<String, List<Rule>> paths = new HashMap<>();
-
-        paths.put(defaultDeclaredPath, new ArrayList<>());
-
-        if (!swaggerDescriptor.isWithPolicyPaths()) {
-            apiEntity.setPaths(paths);
-        }
-
-        if (!swaggerDescriptor.isWithPathMapping()) {
-            apiEntity.setPathMappings(singleton(defaultDeclaredPath));
+        if (pathMappings.isEmpty()) {
+            final String defaultDeclaredPath = "/";
```

Comment:
What if `swaggerDescriptor.isWithPolicyPaths()` is false ?
It should at least have the defaultDeclaredPath, no ?

Reply:
Not sure I understand. 

But yes the idea is that defaultDeclaredPath is present if there are no path mappings. If I understood the intention of the code before. and therefore also if isWithPolicyPaths is false